lib/Sabberworm/CSS/Parser.php

```diff
@@ -56,7 +59,7 @@ public function __construct($sText, Settings $oParserSettings = null) {
 
 	public function setCharset($sCharset) {
 		$this->sCharset = $sCharset;
-		$this->iLength = $this->strlen($this->sText);
+		$this->iLength = count($this->aText);
```

Comment:
Also, the `@charset` declaration will call this method during parsing. But it should trigger a re-parse if the count is different in the new charset, which it has never done so again, this is due for a rework that hasn’t necessarily got anything to do with your changeset.